### PR TITLE
Feature 54088 json api

### PR DESCRIPTION
**Pull request by TheWalkingLeek**

### CODE REVIEW

File: frontend/app/components/account-form.js [hunk 1]
```diff
@@ -77,30 +73,19 @@ export default class AccountForm extends BaseFormComponent {
...
-    if (isNone(group)){
-      this.selectedGroup = null;
-      this.changeset.groupId = null;
-      this.changeset.group = null;
-    } else {
-      this.selectedGroup = group;
-      this.changeset.groupId = group.id;
-      this.changeset.group = group;
-    }
+    this.changeset.group = group;
```
Comment (lucashabersaat) [inline frontend/app/components/account-form.js:88]: Nice decimation of the whole account-form.js. :axe:

File: app/controllers/api/teams/members_controller.rb [hunk 2]
```diff
@@ -38,13 +38,12 @@ def create
 
       # DELETE /api/teams/:team_id/members/:id
       def destroy
-        authorize team, :remove_member?
-        username = teammember.user.username
+        authorize team, :team_member?
```
[the remark this answers is not in the document]
Comment (TheWalkingLeek) [reply app/controllers/api/teams/members_controller.rb:41]: Yes. That is inside the teammember.rb model as a before_destroy hook :)

File: app/controllers/crud_controller.rb [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    if entry.save
+      render_entry({ status: :created,
+                     location: entry_url }
+                   .merge(options[:render_options] || {}))
+    else
+      render_errors
+    end
+  end
+
+  # PATCH/PUT /users/1
```
Comment (psunix) [inline app/controllers/crud_controller.rb:24]: users ?
/entries/1 ?

File: app/controllers/crud_controller.rb [hunk 4]
```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+# A generic controller to display, create, update and destroy entries of a certain model class.
+class CrudController < ListController
```
Comment (psunix) [inline app/controllers/crud_controller.rb:4]: dry crud :heart_eyes:

File: spec/controllers/api/api_users/token_controller_spec.rb [hunk 5]
```diff
@@ -26,6 +27,12 @@
       expect(api_user.authenticate(old_token)).to eq false
       expect(api_user.authenticate(new_token)).to eq true
     end
+
+    it 'user cannot renew token of foreign_api_user' do
+      get :show, params: { id: foreign_api_user.id }, xhr: true
+
+      expect(response).to have_http_status(403)
```
Comment (psunix) [inline spec/controllers/api/api_users/token_controller_spec.rb:34]: :+1:

File: spec/controllers/api/team/groups_controller_spec.rb [hunk 6]
```diff
@@ -12,9 +12,18 @@
...
       get :index, params: { team_id: team }, xhr: true
 
-      groups = json['data']['groups']
+      attributes = data.first['attributes']
 
-      expect(groups.first['name']).to eq 'group1'
+      expect(attributes['name']).to eq 'group1'
+    end
+
+    it 'does not list all groups of a foreign team' do
```
Comment (psunix) [inline spec/controllers/api/team/groups_controller_spec.rb:20]: does not list groups without team membership ?

**Pull request merged**